docs/posts/overview-of-ballerina-0.982.0.md

```diff
@@ -0,0 +1,325 @@
+messages via different resources to the same channel and can also be used for inter-worker communication and worker 
+synchronization.
+
+Defining a channel constrained by the `JSON` type
+```ballerinalerina
+channel<json> jsonChannel; 
+```
+
+Sending a message to the channel
+    
+```ballerina
+# One of the receivers waiting on the key `key` receives the message.
+# If there is no receiver, the message is stored and execution continues.
+# A receiver can arrive later and fetch the message.
+message -> jsonChannel, key; 
+```
+
+Receive a message from the channel with the given key
+   
+```ballerina
+# Execution waits here if the message is not available.
+json result <- jsonChannel, key;
+```
+
+### Introduction of `abstract` Objects
+
+An abstract object is identified by the `abstract` keyword. It describes the type of each field and each method, but 
+not the implementation of the methods. An abstract object type must not have an object constructor method and does not 
+have an implicit initial value. An abstract object type cannot  be initialized using the object initializer.
+
+```ballerina
+public type Foo abstract object {
+    public string name;
+    public int id;
+
+    function getName() returns string;
+
+    function getID() returns int;
+};
+```
+
+### Introduction of Record Iteration Support
+
+Records are now an iterable type. With this change, the `foreach` statement and iterable operations can now be used 
+with records. When iterating a record, one can iterate either over the fields (i.e., field name and value pair) or only 
+over the field values.
+
+```ballerina
+type Person record {
+    string name;
+    int age;
+    string address;
+};
+```
+
+The `foreach` statement can be used on an instance of this record as follows:
+
+```ballerina
+    foreach field, value in person {
+        io:println(field + " : " + <string>value);
+    }
+```
+
+Or if iterating only through the field values:
+```ballerina
+    foreach value in person {
+        io:println(value);
+    }
+```
+
+### Changes to the `main` function
+
+The `main` function now needs to marked `public` and the `main` function could now return an int.
+ 
+Any public function in the entry package could now be invoked via `ballerina run`.
+For example, the public function `add` in the `calculator` package could be invoke, with the integer arguments `4` and 
+`5`, via `ballerina run` as follows:
+
+```cmd
+ballerina run calculator:add 4 5
+```
+If no function is specified, the name of the function to run is considered `main`.
+
+The function invoked via `ballerina run` (including the `main` function) would be data-binding and could have zero or 
+more parameters of any supported type, including any number of required/defaultable parameters and a single rest 
+parameter. This function could also return a value.
+
+For example, given the following public function in the package `currency`:
```

Comment:
Minor change:
"For example, consider the following public function in the `currency` package:"